_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,338 @@
+---
+
+For sure you have heard about Elasticsearch or the Elastic Stack. It seems it started as a search engine based on Lucene, an open-source search engine library built by Shay Banon to index his wife's cooking recipes. From the early days, Elasticsearch has gone a long way and evolved into the Elastic Stack, a great suite for taking data from any source, search, analyze and visualize it in near real-time. So Elasticsearch is a distributed document store, data is serialized as JSON documents and stored distributed across cluster nodes. The _inverted index_ is a data structure that lists every unique word that appears in any document, and also the documents each word appears in. This gives support to fast full-text searches, a feature not supported or partially supported by database engines. The index is a collection of documents and each document is a collection of fields. In turn, each field can be indexed in an optimal data structure, for example, an inverted index for text fields, but a BKD tree for a numeric field.
+
+How can be Elasticsearch integrated into a Spring Boot Application? What are the options for adding Elasticsearch to a Java application? This post will give you a quick introduction to Elasticsearch integration options.
+
+**Prerequisites**:
+
+- [HTTPie](https://httpie.io/)
+- [JHipster 7](https://www.jhipster.tech/installation/)
+- [Java 11+](https://openjdk.java.net/install/index.html)
+- [Okta CLI](https://cli.okta.com)
+- [Docker](https://docs.docker.com/engine/install/)
+- [Docker Compose](https://docs.docker.com/compose/install/)
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+## Elasticsearch integration options for Spring Boot
+
+For Java applications including Spring Boot applications, Elasticsearch provides the following [clients](https://www.elastic.co/guide/en/elasticsearch/client/index.html) for integration:
+
+- [**Java Transport Client**](https://www.elastic.co/guide/en/elasticsearch/client/java-api/current/index.html) Deprecated in Elasticsearch 7.0.0. Provides a client object to execute all operations asynchronously, accepting a listener or returning a future.
+
+- [**Java REST Client**](https://www.elastic.co/guide/en/elasticsearch/client/java-rest/7.15/index.html) Composed by the Low Level REST Client and the High Level REST Client. The Low Level Client provides load balancing, failover, persistent connections, and request/response trace logging. The High Level Client works on top of the Low Level Client and is the replacement for the `TransportClient`. It depends on the Elasticsearch core and provides synchronous and asynchronous APIs.
+
+- [**Java API Client**](https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/current/index.html) The new client library, independent from Elasticsearch core, provides strongly typed requests and responses, blocking and asynchronous versions for all APIs, fluent builders and functional patterns, jackson and JSON-b support.
```

Comment:
Please remove bold from links as we like to have their styles be consistent between posts. I'd also suggest adding a dash or colon after "Client".

Reply:
Done